test/spec/features/service/exec_spec.rb

```diff
@@ -47,6 +47,7 @@
         end
       end
       expect(k.run).to be_truthy
+      expect(k.out).to match /lib64/
```

Comment:
Something fishy going on here. The test spec was passing even though the output was just ` [error] connection closed with code 4000: ArgumentError: width and height must be integers > 0`... is the CLI not exiting with an error?

Reply:
Seems like `expect(k.run).to be_truthy` is not enough... you need `expect(k.code).to be_zero`?

Reply:
Fixed in the e2e specs to actually have these exec specs fail.

```
service exec
  runs a command inside a service
  exits with error if command fails
  exits with command error
  returns an error if command not found
  --tty
    runs a command inside a service (FAILED - 1)
    runs a command with tty control input (FAILED - 2)
    runs a command with non-ascii input (FAILED - 3)
    runs a command with chunked UTF-8 input (PENDING: No reason given)
  --interactive
    runs a command with piped stdin
    runs a command with piped non-ascii stdin
  with multiple instances
    runs a command inside a service on a given instances
    runs a command on every instance with --all
    fails early if running a command on every instances with --all fails
    keeps going if running a command on every instances with --all --skip

container exec
  executes command in a given container
  exits with error if command fails
  exits with command error
  fails if container does not exist
  runs a command inside a container with tty (FAILED - 4)

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) service exec --tty runs a command with chunked UTF-8 input
     # No reason given
     Failure/Error: expect(k.code).to be_zero, k.out
        [error] connection closed with code 4000: ArgumentError: width and height must be integers > 0
     # ./spec/features/service/exec_spec.rb:104:in `block (3 levels) in <top (required)>'

Failures:

  1) service exec --tty runs a command inside a service
     Failure/Error: expect(k.code).to be_zero, k.out
        [error] connection closed with code 4000: ArgumentError: width and height must be integers > 0
     # ./spec/features/service/exec_spec.rb:51:in `block (3 levels) in <top (required)>'

  2) service exec --tty runs a command with tty control input
     Failure/Error: expect(k.out).to match /\^C/
     
       expected " [\e[31merror\e[0m] connection closed with code 4000: ArgumentError: width and height must be integers > 0\n" to match /\^C/
       Diff:
       @@ -1,2 +1,2 @@
       -/\^C/
       + [error] connection closed with code 4000: ArgumentError: width and height must be integers > 0
       
     # ./spec/features/service/exec_spec.rb:67:in `block (3 levels) in <top (required)>'

  3) service exec --tty runs a command with non-ascii input
     Failure/Error: expect(k.code).to be_zero, k.out
        [error] connection closed with code 4000: ArgumentError: width and height must be integers > 0
     # ./spec/features/service/exec_spec.rb:81:in `block (3 levels) in <top (required)>'

  4) container exec runs a command inside a container with tty
     Failure/Error: expect(k.code).to be_zero, k.out
        [error] connection closed with code 4000: ArgumentError: width and height must be integers > 0
     # ./spec/features/container/exec_spec.rb:49:in `block (2 levels) in <top (required)>'

Finished in 53.14 seconds (files took 0.55529 seconds to load)
19 examples, 4 failures, 1 pending

Failed examples:

rspec ./spec/features/service/exec_spec.rb:41 # service exec --tty runs a command inside a service
rspec ./spec/features/service/exec_spec.rb:54 # service exec --tty runs a command with tty control input
rspec ./spec/features/service/exec_spec.rb:70 # service exec --tty runs a command with non-ascii input
rspec ./spec/features/container/exec_spec.rb:37 # container exec runs a command inside a container with tty
```